src/core/shelf.sol

```diff
@@ -103,19 +105,21 @@ contract Shelf is TitleOwned {
         move(loan, shelf[loan].registry, shelf[loan].tokenId, usr);
         adjust(loan);
     }
-
-    function free(uint loan, address usr) public auth  {
-        NFTLike(shelf[loan].registry).transferFrom(address(this), usr, shelf[loan].tokenId);
-        adjust(loan);
-    }
-
     function deposit (uint loan, address usr) public owner(loan) {
         NFTLike(shelf[loan].registry).transferFrom(usr, address(this), shelf[loan].tokenId);
         pile.borrow(loan, shelf[loan].principal);
         shelf[loan].principal = 0;
         adjust(loan);
     }
 
+    // Used by the collector
+    function claim(uint loan, address usr) public auth {
+        NFTLike(shelf[loan].registry).transferFrom(address(this), usr, shelf[loan].tokenId);
```

Comment:
Why did you remove the adjust?

We have two options either we call adjust after each interaction which changes the bags or we don't guarantee an updated bags value and each external call needs to call adjust before.